code/obj/item/clothing/helmets.dm

```diff
@@ -116,30 +116,81 @@
 	desc = "Helps protect against vacuum. Comes in a unique, flashy style."
 
 /obj/item/clothing/head/helmet/space/custom
+
 	name = "bespoke space helmet"
-	desc = "Helps protect against vacuum, and is custom-made just for you!"
-	onMaterialChanged()
-		if(src.material)
-			if(material.hasProperty("thermal"))
-				var/prot = round((100 - material.getProperty("thermal")) / 2)
+	desc = "A custom built helmet with a visor that encompasses most of the face."
+	icon_state = "spacemat"
+
+	inhand_image_icon = 'icons/mob/inhand/hand_headgear.dmi' // inhand shit
+	item_state = "s_helmet"
+
+	icon = 'icons/obj/clothing/item_hats.dmi' // prep the world icon_state for building, is made later
+	var/datum/material/fabr_material = null
+	var/datum/material/visr_material = null
+	var/image/fabrImg = null
+	var/image/visrImg = null
+
+	New()
+		..()
+		fabrImg = image(icon, "spacemat")
+		visrImg = image(icon, "spacemat-vis")
+		overlays += fabrImg
+		overlays += visrImg
+
+	proc/setupVisorMat(var/datum/material/V, var/datum/material/C)
+		src.color = C.color
+		visr_material = copyMaterial(V)
+		if (V)
+			if (V.hasProperty("thermal"))
+				var/prot = round((100 - V.getProperty("thermal")) / 2)
 				setProperty("coldprot", 10+prot)
 				setProperty("heatprot", 1+round(prot/2))
 			else
 				setProperty("coldprot", 10)
 				setProperty("heatprot", 2)
 
-			if(material.hasProperty("permeable"))
-				var/prot = 100 - material.getProperty("permeable")
+			if (V.hasProperty("permeable"))
+				var/prot = 100 - V.getProperty("permeable")
 				setProperty("viralprot", prot)
 			else
 				setProperty("viralprot", 40)
 
-			if(material.hasProperty("density"))
-				var/prot = round(material.getProperty("density") / 20)
+			if (V.hasProperty("density"))
+				var/prot = round(V.getProperty("density") / 20)
 				setProperty("meleeprot_head", 2+prot)
 			else
 				setProperty("meleeprot_head", 2)
 
+	proc/setVisrMaterial(var/datum/material/M)
+		visr_material = copyMaterial(M)
+		overlays -= visrImg
+		if (M)
+			visrImg.color = M.color
+			visrImg.alpha = M.alpha
+		else
+			visrImg.color = null
+			visrImg.alpha = 255
+		overlays += visrImg
+		setName()
+
+	proc/setFabrMaterial(var/datum/material/M)// since color already initalized in setup visor mat, only take alpha
+		fabr_material = copyMaterial(M)
+		src.setMaterial(fabr_material,copy = 0, appearance = 0, setname = 0)
+		overlays -= fabrImg
+		if (M)
+			fabrImg.alpha = M.alpha
+		else
+			fabrImg.color = null
+			fabrImg.alpha = 255
+		overlays += fabrImg
+		setName()
```

Comment:
Move to `onMaterialUpdate` as you already set the material using `setMaterial`. Use `UpdateOverlays` and `SafeGetOverlayImage`